File: src/main/java/de/zerotask/minecraft/tooltweaks/common/CommonProxy.java
package de.zerotask.minecraft.tooltweaks.common;

import de.zerotask.minecraft.tooltweaks.ToolTweaks;
import de.zerotask.minecraft.tooltweaks.common.handler.BowHandler;
import de.zerotask.minecraft.tooltweaks.common.handler.HoeHandler;
import de.zerotask.minecraft.tooltweaks.common.handler.SwordHandler;
import de.zerotask.minecraft.tooltweaks.common.handler.ToolHandler;
import de.zerotask.minecraft.tooltweaks.common.network.PacketHandler;
import net.minecraftforge.common.MinecraftForge;

public class CommonProxy {

    public void preInit() {
        ToolTweaks.LOG.debug("Initializing packet handler");
        PacketHandler.init();
    }

    public void init() {
        ToolTweaks.LOG.debug("Adding various handlers");
        MinecraftForge.EVENT_BUS.register(new BowHandler());
        MinecraftForge.EVENT_BUS.register(new HoeHandler());
        MinecraftForge.EVENT_BUS.register(new SwordHandler());
        MinecraftForge.EVENT_BUS.register(new ToolHandler());
    }
}

File: src/main/kotlin/de/zerotask/minecraft/tooltweaks/common/CommonProxy.kt
package de.zerotask.minecraft.tooltweaks.common

import de.zerotask.minecraft.tooltweaks.common.handler.BowHandler
import de.zerotask.minecraft.tooltweaks.common.handler.HoeHandler
import de.zerotask.minecraft.tooltweaks.common.handler.SwordHandler
import de.zerotask.minecraft.tooltweaks.common.handler.ToolHandler
import de.zerotask.minecraft.tooltweaks.common.network.PacketHandler
import net.minecraftforge.common.MinecraftForge

open class CommonProxy {

    open fun preInit() {
        PacketHandler.init()
    }

    open fun init() {
        MinecraftForge.EVENT_BUS.register(BowHandler())
        MinecraftForge.EVENT_BUS.register(HoeHandler())
        MinecraftForge.EVENT_BUS.register(SwordHandler())
        MinecraftForge.EVENT_BUS.register(ToolHandler())
    }
}